1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,166 @@
+
+Let's take a look at an example of an user providing their own certificate for the HTTP Bridge server:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 443
+    sslConfigurations: #NEW FIELD
+      certificateAndKey: 
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+...
+```
+An example of allowing the operator to issue and manage the certificate where Kafka cluster and KafkaBridge resources are running in the same namespace:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 443
+    sslConfigurations:
+      generateCertficiate: true
+...
+```
+An example of allowing the operator to issue and manage the certificate where Kafka cluster and KafkaBridge resources are running in different namespaces:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 443
+    sslConfigurations:
+      generateCertficiate: true
+      caNamespace: kafka
+...
+```
+
+Users can also configure the certificate validity days and renewal days instead of using CA's defaults by setting these optional fields:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  # HTTP configuration (required)
+  http:
+    port: 443
+    sslConfigurations:
+      generateCertficiate: true
+      certificateValidityDays: 90 #CA default is 365
+      certificateRenewalDays:  14 #CA default is 30
+...
+```
+
+Users can also configure the secure transport protocols by setting these additional fields that are optional:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 443
+    sslConfigurations:
+      sslProtocol: "TLSv1.3" #This will add the protocol to the default list of enabled protocols.
+...
+```
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  # HTTP configuration (required)
+  http:
+    port: 443
+    sslConfigurations:
+      # This will override the default list of enabled protocols.
+      # As of Vertx version 5.0.3, the default list is  "TLSv1, TLSv1.1, TLSv1.2, TLSv1.3".
+      sslEnabledProtocols: "TLSv1.2,TLSv1.3" 
```

Comment:
I understand we probably want dedicated API for seleting the keystore. But do we want it also for all the other options? Why not keep things such as protocol configuration for a regular `config` section?

Reply:
So that it can be used for Kafka clients as well?

Reply:
No, I'm more concerned that overtime you would have lot of different fields here in the API with different options just to cover everyones requirement. Having it in ther `config` makes this easier as that is just a `Map`.